query/src/main/java/org/infinispan/query/dsl/embedded/impl/jpalucene/HibernateSearchPropertyHelper.java

```diff
@@ -0,0 +1,279 @@
+import org.hibernate.search.bridge.builtin.NumericEncodingDateBridge;
+import org.hibernate.search.bridge.builtin.NumericFieldBridge;
+import org.hibernate.search.bridge.builtin.StringEncodingCalendarBridge;
+import org.hibernate.search.bridge.builtin.StringEncodingDateBridge;
+import org.hibernate.search.bridge.builtin.impl.NullEncodingTwoWayFieldBridge;
+import org.hibernate.search.bridge.builtin.impl.TwoWayString2FieldBridgeAdaptor;
+import org.hibernate.search.engine.metadata.impl.DocumentFieldMetadata;
+import org.hibernate.search.engine.metadata.impl.EmbeddedTypeMetadata;
+import org.hibernate.search.engine.metadata.impl.PropertyMetadata;
+import org.hibernate.search.engine.metadata.impl.TypeMetadata;
+import org.hibernate.search.engine.spi.DocumentBuilderIndexedEntity;
+import org.hibernate.search.engine.spi.EntityIndexBinding;
+import org.hibernate.search.spi.SearchIntegrator;
+import org.infinispan.objectfilter.impl.hql.ObjectPropertyHelper;
+import org.infinispan.query.logging.Log;
+import org.jboss.logging.Logger;
+
+import java.text.ParseException;
+import java.util.Calendar;
+import java.util.List;
+
+/**
+ * @author anistor@redhat.com
+ * @since 9.0
+ */
+public class HibernateSearchPropertyHelper extends ObjectPropertyHelper<Class<?>> {
+
+   private static final Log log = Logger.getMessageLogger(Log.class, HibernateSearchPropertyHelper.class.getName());
+
+   private final SearchIntegrator searchFactory;
+
+   private final JPALuceneTransformer.FieldBridgeProvider fieldBridgeProvider;
+
+   public HibernateSearchPropertyHelper(SearchIntegrator searchFactory, EntityNamesResolver entityNamesResolver, JPALuceneTransformer.FieldBridgeProvider fieldBridgeProvider) {
+      super(entityNamesResolver);
+      this.searchFactory = searchFactory;
+      this.fieldBridgeProvider = fieldBridgeProvider;
+   }
+
+   @Override
+   public Class<?> getEntityMetadata(String targetTypeName) {
+      return entityNamesResolver.getClassFromName(targetTypeName);
+   }
+
+   /**
+    * Returns the given value converted into the type of the given property as determined via the field bridge of the
+    * property.
+    *
+    * @param value        the value to convert
+    * @param entityType   the type hosting the property
+    * @param propertyPath the name of the property
+    * @return the given value converted into the type of the given property
+    */
+   @Override
+   public Object convertToPropertyType(String entityType, List<String> propertyPath, String value) {
+      String[] path = propertyPath.toArray(new String[propertyPath.size()]);
+      FieldBridge bridge = getFieldBridge(entityType, path);
+      return convertToPropertyType(value, bridge);
+   }
+
+   @Override
+   public Class<?> getPrimitivePropertyType(String entityType, String[] propertyPath) {
+/* TODO [anistor] XMember is always null. There has to be another way to access hibernate search property metadata...
```

Comment:
This code path is not currently invoked (and the code does not really work). It will be when ISPN-6436 is fixed.